src/service/bind-impl.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {fromClassForDoc} from '../service';
+import * as ngExpressions from "angular-expressions";
+
+export class BindService {
+  constructor(ampdoc) {
+    this.ampdoc = ampdoc;
+    // Map<string, any>
+    this.scope_ = {};
+    this.expressions_ = {};
+  }
+
+  /**
+   * Function that add customized jsonObj to the bindService data set.
+   * @param {!Object} newData
+   */
+  addData(newData) {
+    for (var key in newData) {
```

Comment:
`Object.assign` will be quicker, unless you specifically want to include the prototype's inherited properties.